runtime/src/main/java/io/syndesis/integration/runtime/stephandlers/FunctionHandler.java

```diff
@@ -13,38 +13,67 @@
  */
 package io.syndesis.integration.runtime.stephandlers;
 
+import java.util.Map;
+import java.util.stream.Collectors;
+
 import com.google.auto.service.AutoService;
 import io.syndesis.integration.model.steps.Function;
 import io.syndesis.integration.model.steps.Step;
 import io.syndesis.integration.runtime.StepHandler;
 import io.syndesis.integration.runtime.SyndesisRouteBuilder;
-import io.syndesis.integration.support.Strings;
+import org.apache.camel.CamelContext;
+import org.apache.camel.NoTypeConversionAvailableException;
+import org.apache.camel.TypeConverter;
 import org.apache.camel.model.ProcessorDefinition;
+import org.apache.camel.util.ObjectHelper;
 
 @AutoService(StepHandler.class)
 public class FunctionHandler implements StepHandler<Function> {
+    @Override
+    public boolean canHandle(Step step) {
+        return step.getClass().equals(Function.class);
+    }
+
+    @Override
+    public ProcessorDefinition handle(Function step, ProcessorDefinition route, SyndesisRouteBuilder routeBuilder) {
+        final CamelContext context = routeBuilder.getContext();
+        final TypeConverter converter = context.getTypeConverter();
+
+        String method = null;
+        String function = step.getName();
+        String options = null;
+
+        int idx = function.indexOf("::");
+        if (idx > 0) {
+            method = function.substring(idx + 2);
```

Comment:
what happens for a function name like "bla::" ? Then this leads to an IndexOutOfBoundException.